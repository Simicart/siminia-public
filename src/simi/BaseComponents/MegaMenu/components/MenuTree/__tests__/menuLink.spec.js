import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import MenuLink from '../menuLink';

jest.mock('../toggleTrigger.js', () => 'ToggleTrigger');
jest.mock('@magento/venia-drivers', () => ({
  __esModule: true,
  Link: props => <mock-Link>{props.children}</mock-Link>
}));

const defaultProps = {
  id: 5,
  url: 'venia-bottoms.html',
  name: 'bottoms',
  label: 'Sale',
  label_text_color: '#fff',
  label_background_color: '#000',
  icon: '/icon.png',
  subCategories: [{ id: 3 }],
  toggleAction: jest.fn().mockName('toggleAction'),
  view: 'hamburger',
  level: 1,
  isOpen: false,
  isShowIcons: false,
  onNavigate: jest.fn().mockName('onNavigate')
};

test('renders correct link', () => {
  const tree = createTestInstance(<MenuLink {...defaultProps} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders without label', () => {
  const tree = createTestInstance(<MenuLink {...defaultProps} label={null} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders without toggle btn', () => {
  const tree = createTestInstance(
    <MenuLink {...defaultProps} subCategories={null} />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders without icon', () => {
  const tree = createTestInstance(<MenuLink {...defaultProps} icon={null} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
