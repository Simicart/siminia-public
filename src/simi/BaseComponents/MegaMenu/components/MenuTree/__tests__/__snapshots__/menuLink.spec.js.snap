// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct link 1`] = `
<div
  className="  "
>
  <mock-Link>
    <span>
      bottoms
    </span>
    <span
      style={
        Object {
          "--am-mega-menu-label-bg": "#000",
          "--am-mega-menu-label-color": "#fff",
        }
      }
    >
      Sale
    </span>
  </mock-Link>
  <ToggleTrigger
    action={[Function]}
    isOpen={false}
  />
</div>
`;

exports[`renders without icon 1`] = `
<div
  className="  "
>
  <mock-Link>
    <span>
      bottoms
    </span>
    <span
      style={
        Object {
          "--am-mega-menu-label-bg": "#000",
          "--am-mega-menu-label-color": "#fff",
        }
      }
    >
      Sale
    </span>
  </mock-Link>
  <ToggleTrigger
    action={[Function]}
    isOpen={false}
  />
</div>
`;

exports[`renders without label 1`] = `
<div
  className="  "
>
  <mock-Link>
    <span>
      bottoms
    </span>
  </mock-Link>
  <ToggleTrigger
    action={[Function]}
    isOpen={false}
  />
</div>
`;

exports[`renders without toggle btn 1`] = `
<div
  className="  "
>
  <mock-Link>
    <span>
      bottoms
    </span>
    <span
      style={
        Object {
          "--am-mega-menu-label-bg": "#000",
          "--am-mega-menu-label-color": "#fff",
        }
      }
    >
      Sale
    </span>
  </mock-Link>
</div>
`;
